tools/cldr-code/src/main/java/org/unicode/cldr/json/Ldml2JsonConverter.java

```diff
@@ -376,12 +376,7 @@ private String transformPath(final String pathStr, final String pathPrefix) {
                 activeNumberingSystems.add(ns);
             }
         }
-        DtdType fileDtdType;
-        if (CLDRFile.isSupplementalName(locID)) {
-            fileDtdType = DtdType.supplementalData;
-        } else {
-            fileDtdType = DtdType.ldml;
-        }
```

Comment:
This was the root issue, an incomplete if-tree. Better to use CLDRFile.getDtdType().